File: GreenFlagAssignment1.5/app/src/main/java/com/example/hanad/greenflagassignment1/Old/PersonAdapter.java
//package com.example.hanad.greenflagassignment1;
//
//import android.content.Context;
//import android.support.v7.widget.RecyclerView;
//import android.view.LayoutInflater;
//import android.view.View;
//import android.view.ViewGroup;
//import android.widget.ImageView;
//import android.widget.RadioButton;
//import android.widget.Spinner;
//import android.widget.TextView;
//import android.widget.Toast;
//
//import com.example.hanad.greenflagassignment1.model.CustomerModel;
//import com.squareup.picasso.Picasso;
//
//import java.util.ArrayList;
//
///**
// * Created by hanad on 05/02/2018.
// *
// * Steps to follow:
// * 1. initialize the objects in the constructor
// * 2. Create Innter ViewHolder class
// * 2.1. MyViewHolder extend RecyclerView.ViewHolder
// */
//
//class PersonAdapter extends RecyclerView.Adapter<PersonAdapter.MyViewHolder>{
//
//    private Context applicationContext;
//    private int row_customer;
//    private  ArrayList<CustomerModel> arrayList;
//
//    public PersonAdapter(Context applicationContext, int row_customer, ArrayList<CustomerModel> arrayList) {
//        this.applicationContext = applicationContext;
//        this.arrayList = arrayList;
//        this.row_customer = row_customer;
//    }
//
//    /**
//     * Load the row layout
//     * @param parent
//     * @param viewType
//     * @return
//     */
//    @Override
//    public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
//        return new MyViewHolder(LayoutInflater.from(parent.getContext()).inflate(row_customer,parent,false));
//
//    }
//
//    /**
//     * Load data into individual UI
//     * @param holder
//     * @param position
//     */
//    @Override
//    public void onBindViewHolder(MyViewHolder holder, int position) {
//        holder.mName.setText(arrayList.get(position).getName());
//        holder.mUsername.setText(arrayList.get(position).getUsername());
//        holder.mPassword.setText(arrayList.get(position).getPassword());
//        holder.mAge.setText(arrayList.get(position).getAge());
//        holder.mBirthday.setText(arrayList.get(position).getBirthday());
//        holder.mCountry.setText(arrayList.get(position).getCountry());
//        holder.mRadio.setText(arrayList.get(position).getRadio());
//
//        Picasso.with(applicationContext)
//                .load(arrayList.get(position).getImage())
//                .resize(100, 100)
//                .centerCrop()
//                .into(holder.mImageView);
//
//        holder.callItemClick(new ItemClickListener() {
//            @Override
//            public void OnClick(View view, int position) {
//                Toast.makeText(applicationContext, arrayList.get(position).getName(),Toast.LENGTH_LONG).show();
//                Toast.makeText(applicationContext, arrayList.get(position).getUsername(),Toast.LENGTH_LONG).show();
//                Toast.makeText(applicationContext, arrayList.get(position).getPassword(),Toast.LENGTH_LONG).show();
//                Toast.makeText(applicationContext, arrayList.get(position).getAge(),Toast.LENGTH_LONG).show();
//                Toast.makeText(applicationContext, arrayList.get(position).getBirthday(),Toast.LENGTH_LONG).show();
//                Toast.makeText(applicationContext, arrayList.get(position).getRadio(),Toast.LENGTH_LONG).show();
//            }
//        });
//    }
//
//    @Override
//    public int getItemCount() {
//        return arrayList.size();
//
//    }
//
//    public class MyViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
//
//        private TextView mName;
//        private TextView mUsername;
//        private TextView mPassword;
//        private TextView mAge;
//        private TextView mBirthday;
//        private ImageView mImageView;
//        private TextView mCountry;
//        private TextView mRadio;
//
//
//
//        private ItemClickListener itemClickListener;
//
//        public MyViewHolder(View itemView) {
//            super(itemView);
//            mName = itemView.findViewById(R.id.mName);
//            mUsername = itemView.findViewById(R.id.mUsername);
//            mPassword = itemView.findViewById(R.id.mPassword);
//            mAge = itemView.findViewById(R.id.mAge);
//            mBirthday = itemView.findViewById(R.id.mBirthday);
//            mImageView = itemView.findViewById(R.id.mImageView);
//            mCountry = itemView.findViewById(R.id.mCountry);
//            mRadio = itemView.findViewById(R.id.mRadio);
//
//
//
//
//            itemView.setOnClickListener(this);
//        }
//
//    public void callItemClick(ItemClickListener itemClickListener){
//        this.itemClickListener=itemClickListener;
//    }
//        @Override
//        public void onClick(View view) {
//            itemClickListener.OnClick(view,getPosition());
//        }
//    }
//}

File: GreenFlagAssignment1.5/app/src/main/java/com/example/hanad/greenflagassignment1/PersonAdapter.kt
package com.example.hanad.greenflagassignment1

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RadioButton
import android.widget.Spinner
import android.widget.TextView
import android.widget.Toast

import com.example.hanad.greenflagassignment1.model.CustomerModel
import com.squareup.picasso.Picasso

import java.util.ArrayList

/**
 * Created by hanad on 05/02/2018.
 *
 * Steps to follow:
 * 1. initialize the objects in the constructor
 * 2. Create Innter ViewHolder class
 * 2.1. MyViewHolder extend RecyclerView.ViewHolder
 */

internal class PersonAdapter(private val applicationContext: Context, private val row_customer: Int, private val arrayList: ArrayList<CustomerModel>) : RecyclerView.Adapter<PersonAdapter.MyViewHolder>() {

    /**
     * Load the row layout
     * @param parent
     * @param viewType
     * @return
     */
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(row_customer, parent, false))

    }

    /**
     * Load data into individual UI
     * @param holder
     * @param position
     */
    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.mName.text = arrayList[position].name
        holder.mUsername.text = arrayList[position].username
        holder.mPassword.text = arrayList[position].password
        holder.mAge.text = arrayList[position].age
        holder.mBirthday.text = arrayList[position].birthday
        holder.mCountry.text = arrayList[position].country
        holder.mRadio.text = arrayList[position].radio

        Picasso.with(applicationContext)
                .load(arrayList[position].image)
                .resize(100, 100)
                .centerCrop()
                .into(holder.mImageView)

        holder.callItemClick(ItemClickListener { view, position ->
            Toast.makeText(applicationContext, arrayList[position].name, Toast.LENGTH_LONG).show()
            Toast.makeText(applicationContext, arrayList[position].username, Toast.LENGTH_LONG).show()
            Toast.makeText(applicationContext, arrayList[position].password, Toast.LENGTH_LONG).show()
            Toast.makeText(applicationContext, arrayList[position].age, Toast.LENGTH_LONG).show()
            Toast.makeText(applicationContext, arrayList[position].birthday, Toast.LENGTH_LONG).show()
            Toast.makeText(applicationContext, arrayList[position].radio, Toast.LENGTH_LONG).show()
        })
    }

    override fun getItemCount(): Int {
        return arrayList.size

    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        internal val mName: TextView
        internal val mUsername: TextView
        internal val mPassword: TextView
        internal val mAge: TextView
        internal val mBirthday: TextView
        internal val mImageView: ImageView
        internal val mCountry: TextView
        internal val mRadio: TextView


        private var itemClickListener: ItemClickListener? = null

        init {
            mName = itemView.findViewById(R.id.mName)
            mUsername = itemView.findViewById(R.id.mUsername)
            mPassword = itemView.findViewById(R.id.mPassword)
            mAge = itemView.findViewById(R.id.mAge)
            mBirthday = itemView.findViewById(R.id.mBirthday)
            mImageView = itemView.findViewById(R.id.mImageView)
            mCountry = itemView.findViewById(R.id.mCountry)
            mRadio = itemView.findViewById(R.id.mRadio)




            itemView.setOnClickListener(this)
        }

        fun callItemClick(itemClickListener: ItemClickListener) {
            this.itemClickListener = itemClickListener
        }

        override fun onClick(view: View) {
            itemClickListener!!.OnClick(view, position)
        }
    }
}
